tools/genesis/genesis.go

```diff
@@ -205,18 +207,30 @@ type MultisigDefs struct {
 	MultisigAliaseas    []genesis.UnparsedMultisigAlias
 }
 
-func newMultisigAlias(txID ids.ID, addrs []ids.ShortID, threshold uint32, memo string) (genesis.MultisigAlias, error) {
-	utils.Sort(addrs)
+func newMultisigAlias(txID ids.ID, pks []*secp256k1.PublicKey, threshold uint32, memo string) (genesis.MultisigAlias, error) {
+	publicKeys := make([]multisig.PublicKey, len(pks))
+	for i, pk := range pks {
+		err := publicKeys[i].FromBytes(pk.SerializeCompressed())
+		if err != nil {
+			return genesis.MultisigAlias{}, err
+		}
+	}
+	utils.Sort(publicKeys)
+
 	ma := genesis.MultisigAlias{
-		Threshold: threshold,
-		Addresses: addrs,
-		Memo:      memo,
+		Threshold:  threshold,
+		PublicKeys: publicKeys,
+		Memo:       memo,
 	}
 	ma.Alias = ma.ComputeAlias(txID)
 	return ma, msigSanityCheck(ma, txID)
 }
 
 func msigSanityCheck(ma genesis.MultisigAlias, txID ids.ID) error {
+	if _, err := genesis.MultisigAliasDefFromConfig(ma); err != nil {
+		return err
+	}
+
```

Comment:
`if err := msigAlias.Verify(); err != nil`
where msigAlias is 1nd return from MultisigAliasDefFromConfig